subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -83,7 +158,16 @@ static int package_uri_write_cb(u16_t obj_inst_id,
 {
 	SYS_LOG_DBG("PACKAGE_URI WRITE: %s", package_uri);
 #ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
-	lwm2m_firmware_start_transfer(package_uri);
+	u8_t state = lwm2m_firmware_get_update_state();
+	if (STATE_IDLE == state) {
+		lwm2m_firmware_set_update_result(RESULT_DEFAULT);
+		lwm2m_firmware_start_transfer(package_uri);
+	} else if (STATE_DOWNLOADED == state) {
+		if (data_len == 0) {
```

Comment:
Figure 29 seems to be a possible implementation indeed, but that doesn't cover the cancel update case, which is why my interpretation was more to reset at any point in time, at any state.

Looks like something to ask oma for clarification.